src/cmd/hotspots/transfer.rs

```diff
@@ -0,0 +1,136 @@
+use crate::{
+    cmd::{api_url, get_password, get_txn_fees, load_wallet, Opts},
+    keypair::PubKeyBin,
+    result::Result,
+    traits::{Sign, ToJson, TxnEnvelope, TxnFee, B58, B64},
+};
+use helium_api::{BlockchainTxn, BlockchainTxnTransferHotspotV1, Client, Hnt, Txn};
+use std::io;
+use structopt::StructOpt;
+
+/// Transfer hotspot as buyer or seller.
+#[derive(Debug, StructOpt)]
+pub enum Transfer {
+    Buy(Buy),
+    Sell(Sell),
+}
+
+#[derive(Debug, StructOpt)]
+pub struct Sell {
+    /// Public address of gateway to be transferred
+    gateway: String,
+    /// The recipient of the gateway transfer
+    buyer: String,
+    /// Price in HNT to be paid by recipient of transfer
+    price: Option<Hnt>,
+}
+
+#[derive(Debug, StructOpt)]
+pub struct Buy {
+    /// Base64 encoded transaction to sign. If no transaction if given
+    /// stdin is read for the transaction. Note that the stdin feature
+    /// only works if the wallet password is set in the
+    /// HELIUM_WALLET_PASSWORD environment variable
+    #[structopt(name = "TRANSACTION")]
+    txn: Option<String>,
+    #[structopt(long)]
+    /// Commit to sign and submit the transaction
+    commit: bool,
+}
+
+impl Buy {
+    fn read_txn(&self) -> Result<String> {
+        match &self.txn {
+            Some(txn) => Ok(txn.to_string()),
+            None => {
+                let mut buffer = String::new();
+                io::stdin().read_line(&mut buffer)?;
+                Ok(buffer.trim().to_string())
+            }
+        }
+    }
+}
+
+impl Transfer {
+    pub fn run(self, opts: Opts) -> Result {
+        let wallet = load_wallet(opts.files)?;
+        let client = Client::new_with_base_url(api_url());
+
+        match self {
+            Self::Sell(sell) => {
+                let seller = wallet.address_as_vec();
+                let buyer = PubKeyBin::from_b58(&sell.buyer)?;
+                let buyer_account = client.get_account(&buyer.to_b58()?)?;
+                let gateway = PubKeyBin::from_b58(&sell.gateway)?.to_vec();
+
+                let mut txn = BlockchainTxnTransferHotspotV1 {
+                    fee: 0,
+                    seller,
+                    gateway,
+                    buyer: buyer.to_vec(),
+                    seller_signature: vec![],
+                    buyer_signature: vec![],
+                    amount_to_seller: if let Some(price) = sell.price {
```

Comment:
unwrap_or_else?

Reply:
unwrap here? I left the sell amount optional in the CLI arguements, so if they ommitted it goes to 0. Can enforce its entry instead if thats what you prefer?

Reply:
Ah good point. I have trouble generalizing the unwrap_or_else to the `Option` stuff for some reason.

Reply:
Oh actually I can't use unwrap because when I unwrap I still need the Hnt to Bones conversion.

Reply:
um.. surely you can unwrap_or_else and then call to_bones on the result